internal/factsengine/gatherers/cibadmin_test.go

```diff
@@ -12,28 +12,28 @@ import (
 
 type CibAdminTestSuite struct {
 	suite.Suite
+	executor       *mocks.CommandExecutor
 	cibAdminOutput []byte
 }
 
 func TestCibAdminTestSuite(t *testing.T) {
 	suite.Run(t, new(CibAdminTestSuite))
 }
 
-func (suite *CibAdminTestSuite) SetupSuite() {
+func (suite *CibAdminTestSuite) SetupTest() {
+	suite.executor = new(mocks.CommandExecutor)
 	lFile, _ := os.Open("../../../test/fixtures/gatherers/cibadmin.xml")
```

Comment:
I believe that reading the cibadmin.xml could go in a `SetupSuite()` because reading that file once would be enough, while we need to reset the executor on each test because of the `.On(...)` prophecies (telling how the method should behave).

Yet, this is just fine as well.

Reply:
Indeed, nice catch, this should make the tests execute a bit faster

Reply:
(I don't really get the benefit of the two constructors though :/)

Reply:
@rtorrero In this case the benefit is to have a much more ergonomic external api, because we don't care about having one instance of the executor among all the gatherers (for now, because really the executor is just a function that runs command), in much more advanced scenarios, where the dependency we want to inject needs to be atomic or at least controlled in all the difference instances, the defaul constructor should be omitted

For example a database client with connection parameters and connection pool, we need to inject the dependency explicitly in order to have a coordination and control among the dependency injected to the dependents, so a default constructor can be counter intuitive and lead to confusion 

At the end @rtorrero is up to you, the important thing is that we distinguish this two scenarios when we decide now and in future how to inject the deps

Reply:
Ok, now I see. Indeed in this case, sharing the CommandExecutor instance for all the gatherers work, but I'd also go for what @nelsonkopliku proposes as I think it's just more generic.